textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -59,6 +71,22 @@ public void open() throws DataFlowException {
             TopDocs topDocs = luceneIndexSearcher.search(dataReaderPredicate.getLuceneQuery(), Integer.MAX_VALUE);
             scoreDocs = topDocs.scoreDocs;
             cursor = OPENED;
+
+            this.queryTokens = Utils.tokenizeQuery(dataReaderPredicate.getAnalyzer(),dataReaderPredicate.getQueryString());
+            // sort the query token, as the term vector are also sorted.
+            // This makes the seek faster.
+            this.queryTokens.sort(String.CASE_INSENSITIVE_ORDER);
+
+            this.queryTokensInBytesRef = new ArrayList<>();

```

Comment:
termVerctors are sorted based on the term. That's why I am sorting the query tokens also. So when we search of token in the term vector, the iterator will move only in the forward direction, which should be little faster.